src/pages/index.js

```diff
@@ -1,51 +1,89 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import Fade from 'react-reveal/Fade'
 import classnames from 'classnames'
-
 import Layout from '@theme/Layout'
 import Link from '@docusaurus/Link'
 import useDocusaurusContext from '@docusaurus/useDocusaurusContext'
+import useThemeContext from '@theme/hooks/useThemeContext'
 import useBaseUrl from '@docusaurus/useBaseUrl'
 import styles from './styles.module.css'
 import Translate, { translate } from '@docusaurus/Translate'
 
-// See translations for label and description
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment'
+
+function HeroAnimation() {
+  if (ExecutionEnvironment.canUseDOM) {
+    require('@lottiefiles/lottie-player')
+  }
+
+  const { isDarkTheme } = useThemeContext()
+
+  useEffect(() => {
+    const player = document.querySelector('lottie-player')
+    player.load(
+      isDarkTheme ? 'lottie/tauri-splash.json' : 'lottie/tauri-splash-dark.json'
+    )
+  })
+
+  return (
+    <lottie-player
+      background="transparent"
+      speed="1"
+      style={{ width: '75%', margin: 'auto' }}
+      autoplay
+    ></lottie-player>
+  )
+}
```

Comment:
Not good at react but I think this is better (well, if it works).

useEffect without a dependency array will run the hook with every single update to the component but we need it only to run when the theme changes. 

```suggestion
  const { isDarkTheme } = useThemeContext()
  const  player = useRef(null)
 
 useEffect(() => {
    player.current.load(
      isDarkTheme ? 'lottie/tauri-splash.json' : 'lottie/tauri-splash-dark.json'
    )
  }, [isDarkTheme])

  return (
    <lottie-player
      ref={player}
      background="transparent"
      speed="1"
      style={{ width: '75%', margin: 'auto' }}
      autoplay
    ></lottie-player>
  )
}
```